src/context/filtercontext.jsx

```diff
@@ -0,0 +1,23 @@
+// x
+
+import React, { useContext, createContext, useReducer } from "react";
+import { reducerFunc } from "../reducers/filterReducer";
+
+const FilterContext = createContext();
+const useFilter = () => useContext(FilterContext);
+
+const FilterProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducerFunc, {
```

Comment:
As we are going to have multiple contexts like cart and wishlist, so then we would have muliple states and dispatches, so to differentiate between them I would suggest you to name it something as `[filterState, filterDispatch]`. So there will be no confusion later on